app/models/user.rb

```diff
@@ -2,6 +2,8 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: :author_id
   has_many :likes, foreign_key: :author_id
   has_many :comments, foreign_key: :author_id
+  validates :name, presence: true
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def recent_posts
     posts.order(created_at: :desc).limit(3)
```

Comment:
- [ ] <b>REQUIRED:</b> You did not handle N+1 problem as it is request in project requirements. you can either fix it by adding `include` here or just define it globally in post model like so:

```ruby
default_scope { includes(:comments) }
```

And do the same in comment model

```ruby
default_scope { includes(:likes) }
```

And also in user model

```ruby
default_scope { includes(:posts, :comments, :likes) }
```

Reply:
If I add this I'll get errors from the bullet gem
![routes](https://user-images.githubusercontent.com/63880999/146655768-fdec0761-f025-41c6-95f5-96ca76941191.png)

Reply:
Try this way to see if it can solve the issue you are getting.

`@posts = @user.posts.includes(:comments, :likes).find(params[:id])`

Reply:
OPTIONAL but RECOMMENDED.

- [ ] Add `@posts = @user.posts.includes(:comments, :likes).find(params[:id])` in the posts controller index action to solve the N+1 problem as previously discussed. It has to be done in the controllers and not the models, I bet it's the reason for the error.